common/lib/common-utils/src/bufferBrowser.ts

```diff
@@ -59,7 +62,7 @@ export const bufferToString = (blob: ArrayBufferLike, encoding: string): string
  */
 export function isArrayBuffer(obj: any): obj is ArrayBuffer {
     const maybe = obj as (Partial<ArrayBuffer> & Partial<Uint8Array>) | undefined;
-    return obj instanceof ArrayBuffer
+    return instanceOfArrayBuffer(obj)
```

Comment:
Should we remove this check and leave only duct-typing checks?
It will ensure that duct-typing checks are run for both types of objects (inside and outside of iframe)

Reply:
`instanceOfArrayBuffer(obj)` here will be true for any built-in `ArrayBuffer`, even from other iFrames ...so it's really a [question](https://github.com/microsoft/FluidFramework/pull/6674#discussion_r667217854) of whether or not you had hoped to permit non-authentic (but duck-type approximate) `ArrayBuffer` values here.

I assume permitting authentic built-in `ArrayBuffer`s from other iFrames would be sufficient, and so I'd probably recommend `export const isArrayBuffer = instanceOfArrayBuffer;` here -- or more directly `export const isArrayBuffer = bindInstanceOfBuiltin(new ArrayBuffer(0));`*

---

(*) It doesn't look like this export is getting used anywhere, and the only candidate usage seems to be [in odspDocumentStorageManager.ts](https://github.com/microsoft/FluidFramework/blob/main/packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts#L188), but the exports of `bufferBrowser.ts` not not currently get exported in [index.js](https://github.com/microsoft/FluidFramework/blob/main/common/lib/common-utils/src/index.ts#L7) (through [indexBrowser.js](https://github.com/microsoft/FluidFramework/blob/main/common/lib/common-utils/src/indexBrowser.ts#L6)).  Are `common-utils`/`odsp-driver` not intended to be "isomorphic"?  I'm surprised that `indexNode.ts` gets exposed (but `indexBrowser.ts` is not).

I might recommend a more isomorphic-friendly `export const isArrayBuffer = typeof globalThis.ArrayBuffer === "function" ? bindInstanceOfBuiltin(new ArrayBuffer(0)) : () => false;` (or even a more general `export const isBuffer = typeof globalThis.ArrayBuffer === "function" ? bindInstanceOfBuiltin(new ArrayBuffer(0)) : Buffer.isBuffer;`) that ultimately gets surfaced in `common-utils`'s index.ts.  Thoughts?

Reply:
Both need to be isomorphic. I'm sure both are exported, but I've not looked at disconnect you described.
Sure using instanceOfArrayBuffer() and dropping the rest of duct typing is fine. I think we should chose - either we use duct typing or instanceOfArrayBuffer(). Doing both askes for the other path to almost never be executed, and thus less likely being tested and working when we need it.